### DISCUSSION TITLE
correlation Id

### DISCUSSION THREAD

**Discussion by vejandla** [strapi/strapi#7307] [Legacy Topics]
how can I access ctx globally?

Why? I would like to assign a unique request id to every incoming request and wanted to use that request id whenever I log a request, So I can trace what a specific user is doing as part of the request. 

I already have my authentication middleware which fills claims to the ctx as part of the middleware, I can also add a requestId too. But, I do not want to pass context to my logger function for everycall. Is there a way to access the ctx globally, So I can add the requestId and user info along with every log. 

Hoping I'm not reinventing the wheel here. If so, Please let me know what I should be using for this functionality. 

Thanks in advance.

**Comment by derrickmehaffy** [comment]
Hello,

I believe in this case you will want to tie the data into the `ctx.state.yourkey` which should be passed along through the request chain but I will ping @lauriejim to confirm this. The state is something initialized by Koa and Strapi does set some information there such as `ctx.state.user` (from the users-permissions plugin)

**Reply by vejandla** [reply]
Hi @derrickmehaffy I can certainly do that. but, My question is more towards, how can I access ctx outside my controller? 

If I write my own logger, how can I access this ctx in my logger without explicitly passing it to every log?

**Reply by derrickmehaffy** [reply]
I believe the ctx.state should be carried through the entire request chain without directly passing it, but I want @lauriejim to confirm it as I'm not entirely sure myself. (state I believe is the only one that is passed along iirc)

**Reply by vejandla** [reply]
thanks @derrickmehaffy for the quick reply, I'll wait for @lauriejim 

But, I do not see only strapi is the global object. 

https://github.com/strapi/strapi/blob/master/packages/strapi-generate-new/lib/resources/dot-files/eslintrc#L18

how can I access ctx from `strapi` object.

**Reply by derrickmehaffy** [reply]
You can't directly the two aren't directly related to each other.

Side point as I was looking through other discussions and I do believe that is the case with `ctx.state`: https://github.com/strapi/strapi/discussions/6692#discussioncomment-42620

**Reply by vejandla** [reply]
hmm, probably I might not have presented my question well in the first place. 

I'm planning on accessing a request context across all files (be it controller, or my utility function like logger or simple data transform function).

The problem, I'm running into is, **how to access this context object without passing into chain of function calls**, (ex: controller --> some func --> some other func --> some other func --> .....)

**Reply by derrickmehaffy** [reply]
In that case yeah, you have to actually pass the ctx into it.

ctx.state is used for dynamic data passed between functions
strapi.* is static that is kept at all times but is reinitialized when Strapi restarts
